esrally/track/loader.py

```diff
@@ -1407,29 +1407,41 @@ def parse_parallel(self, ops_spec, ops, challenge_name):
         default_iterations = self._r(ops_spec, "iterations", error_ctx="parallel", mandatory=False)
         default_warmup_time_period = self._r(ops_spec, "warmup-time-period", error_ctx="parallel", mandatory=False)
         default_time_period = self._r(ops_spec, "time-period", error_ctx="parallel", mandatory=False)
+        default_ramp_up_time_period = self._r(ops_spec, "ramp-up-time-period", error_ctx="parallel", mandatory=False)
         clients = self._r(ops_spec, "clients", error_ctx="parallel", mandatory=False)
         completed_by = self._r(ops_spec, "completed-by", error_ctx="parallel", mandatory=False)
 
         # now descent to each operation
         tasks = []
         for task in self._r(ops_spec, "tasks", error_ctx="parallel"):
             tasks.append(self.parse_task(task, ops, challenge_name, default_warmup_iterations, default_iterations,
-                                         default_warmup_time_period, default_time_period, completed_by))
+                                         default_warmup_time_period, default_time_period, default_ramp_up_time_period,
+                                         completed_by))
```

Comment:
the number of positional arguments here is a smell, consider encapsulating in a config dataclass?

Reply:
I agree that this method declares quite some parameters by now. However, given that this is de facto a private method that has only one call site that passes these parameters, I'd not go so far as to introduce a class for that?

Reply:
i was thinking it would be applicable to both this and Task.__init__() but the call sites are mainly dozens of tests, so yes, retracted for now